Source/Core/Core/PowerPC/MMU.h

```diff
@@ -37,49 +37,38 @@ u32 HostRead_Instruction(u32 address);
 std::string HostGetString(u32 address, size_t size = 0);
 
 template <typename T>
-struct TryReadResult
+struct ReadResult
 {
-  // whether the read succeeded; if false, the other fields should not be touched
-  bool success;
-
   // whether the address had to be translated (given address was treated as virtual) or not (given
   // address was treated as physical)
   bool translated;
 
   // the actual value that was read
   T value;
 
-  TryReadResult() : success(false) {}
-  TryReadResult(bool translated_, T&& value_)
-      : success(true), translated(translated_), value(std::move(value_))
-  {
-  }
-  TryReadResult(bool translated_, const T& value_)
-      : success(true), translated(translated_), value(value_)
-  {
-  }
-  explicit operator bool() const { return success; }
+  ReadResult(bool translated_, T&& value_) : translated(translated_), value(std::move(value_)) {}
```

Comment:
I may be wrong here.  But I think since this is a `T&&` this line should be:

`ReadResult(bool translated_, T&& value_) : translated(translated_), value(std::forward<T>(value_)) {}`

The reason is T&& can be either an lvalue or an rvalue.  But move is only applicable on rvalues.

(I know you didn't write this code originally but just thought I'd mention it)

Reply:
Changed.